lightning/src/util/ser_macros.rs

```diff
@@ -1825,4 +1873,33 @@ mod tests {
 		assert_eq!(TuplesOnly::read(&mut none_data_read).unwrap(), None);
 		assert_eq!(none_data_read.position(), unknown_data_variant.len() as u64);
 	}
+
+	#[derive(Debug, PartialEq, Eq)]
+	struct ExpandedField {
+		// Old versions of LDK are presumed to have had something like:
+		// old_field: u8,
+		new_field: (u8, u8),
+	}
+	impl_writeable_tlv_based!(ExpandedField, {
+		(0, old_field, (legacy, u8, {
+			// Sadly the type-checker needs some help
+			let _: &(u8, u8) = &new_field;
+			if let Some(old_field) = old_field {
+				new_field.0 = old_field;
+			}
```

Comment:
I'm a bit confused what a legitimate use case for the `read` closure would be. Do you have an example? It seems like `default_value` accomplishes a lot of its potential goals if a newer field isn't set on read.

Reply:
I'm not following, in the `ChannelDetails` impl for example we copy from `user_channel_id_low` into the new field `user_channel_id`?

Reply:
Ah okay, back to my original question then, is there an example of when we'd want that? Just want to make sure we're not adding extra complexity without a good reason

Reply:
PR description needs updating now.